### PR TITLE
Support building snap images

### PR DESCRIPTION
**Pull request by jwrober**
This is a second attempt from #1713, part of #1674 

Upgrades to use 128x128 icons for higher fidelity. Corrects some license attribution errors.

Should be backported to stable.

What works:
- fc21:// links
- Modpacks downloaded are discovered properly
- Loading the tutorial
- Connected to LT76 just fine
- Playing a local game

Known issues:
- Not related to the snap, but we don't set a user name in the options rc file at "new". Until fixed a user has to go into options and set a user name on network tab
- The top level `icon:` directive seems to override the value for the icons in the .desktop files.
- If you have saved games in `~/.local/share/freeciv21` they won't be found.

### CODE REVIEW

File: dist/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+name: freeciv21
+title: Freeciv21
+summary: Freeciv21 - Freeciv for the 21st Century
+description: |
+  Freeciv21 is a free and open source empire-building strategy game inspired by the history of human civilization. The game commences in prehistory and your mission is to lead your tribe from the Stone Age to the Space Age.
+  This snap package contains everything you need to play local single-player games against the computer as well as connect to online multi-player games against human opponents.
+  Freeciv21 is based on the Qt framework and brought to you by the team at Longturn.net.
+
+# TODO: At some point, will want to integrate into cmake process and configure_file()
+version: '3.0-rc.1'
```
Comment (lmoureaux) [inline dist/snapcraft.yaml:28]: Let's make an issue for this right away.

**Review by lmoureaux — APPROVED** [review]
So setting `FREECIV_DATA_PATH` was not needed in the end?

**Comment by jwrober** [comment]
> So setting `FREECIV_DATA_PATH` was not needed in the end?

Correct. The native paths work fine.

File: dist/snapcraft.yaml (hunk 1 above)
Comment (jwrober) [reply dist/snapcraft.yaml:28]: #1735

**Pull request merged**